chess/__init__.py

```diff
@@ -3817,14 +3819,21 @@ def _repr_svg_(self) -> str:
 
     def __eq__(self, board: object) -> bool:
         if isinstance(board, Board):
-            return (
-                self.halfmove_clock == board.halfmove_clock and
-                self.fullmove_number == board.fullmove_number and
-                type(self).uci_variant == type(board).uci_variant and
-                self._transposition_key() == board._transposition_key())
+            return hash(self) == hash(board)
```

Comment:
`hash(self) == hash(board)` does *not* imply `self` equals `board` (there can be hash collisions).

Reply:
I am not sure I understand the claim. Are you claiming the hash is weak? Because hash collisions are negligible.

Reply:
Maybe for your use case.

But this is a general purpose library. So we shouldn't assume that we will only ever see 2^32 positions (at which point collisions are likely even for strong 64 bit hashes), or even that inputs aren't specifically chosen by an adversary.